### PR TITLE
Rename iterable response

### LINKED ISSUE #4961: Move IterableResponse to appropriate package
IterableResponse is not related to HTTP/REST response type. It is located in the wrong package.

### PR DESCRIPTION
**Pull request by JonathanGiles**
Fixes #4961

### CODE REVIEW

File: sdk/core/azure-core/src/samples/java/com/azure/core/util/IterableStreamJavaDocCodeSnippets.java [hunk 1]
```diff
@@ -33,20 +33,20 @@ public void streamSnippet() throws MalformedURLException {
 
         String deserializedHeaders = "header1,value1,header2,value2";
 
-        IterableResponse<PagedResponseBase<String, Integer>> myIterableResponse =
-            new IterableResponse<>(Flux.just(createPagedResponse(httpRequest, httpHeaders, deserializedHeaders, 1, 3)));
+        IterableStream<PagedResponseBase<String, Integer>> myIterableStream =
+            new IterableStream<>(Flux.just(createPagedResponse(httpRequest, httpHeaders, deserializedHeaders, 1, 3)));
 
-        // BEGIN: com.azure.core.http.rest.iterableResponse.stream
+        // BEGIN: com.azure.core.util.iterableStream.stream
```
Comment (conniey) [inline sdk/core/azure-core/src/samples/java/com/azure/core/util/IterableStreamJavaDocCodeSnippets.java]: I thought the naming would be: `com.azure.core.util.iterablestream.stream`
The guidance suggests that class name should be lowercase. 🤔 But maybe I'm misreading it.

> For other methods in the class: `<packagename>.<classname>.<methodName>`

https://azure.github.io/azure-sdk/java_documentation.html

Comment (JonathanGiles) [reply sdk/core/azure-core/src/samples/java/com/azure/core/util/IterableStreamJavaDocCodeSnippets.java]: Yeah - I'm starting to change my mind here. I think the camel casing helps to make it readable. What I said to Shawn regarding the checkstyle rule was to not enforce case.

Comment (conniey) [reply sdk/core/azure-core/src/samples/java/com/azure/core/util/IterableStreamJavaDocCodeSnippets.java]: I agree the camel casing makes it readable.

Comment (JonathanGiles) [reply sdk/core/azure-core/src/samples/java/com/azure/core/util/IterableStreamJavaDocCodeSnippets.java]: I'll update the spec

**Pull request closed without merging**